src/renderers/dom/shared/wrappers/__tests__/ReactDOMInput-test.js

```diff
@@ -842,6 +842,26 @@ describe('ReactDOMInput', () => {
     );
   });
 
+  it('should warn for type checkbox with a value prop and no checked prop', () => {
+    ReactTestUtils.renderIntoDocument(
+      <input type="checkbox" checked={true} onChange={() => {}} />,
+    );
+    expectDev(console.error.calls.count()).toBe(0);
```

Comment:
We have a helper for this now. For an example, check out: https://github.com/facebook/react/blob/725e499cfb4da60da46aa5b44c4bad29cad3d08f/packages/react-dom/src/__tests__/ReactDOMInput-test.js#L74-L80